dask/dataframe/groupby.py

```diff
@@ -2102,6 +2105,8 @@ def corr(self, ddof=1, split_every=None, split_out=1, numeric_only=no_default):
         """Groupby correlation:
         corr(X, Y) = cov(X, Y) / (std_x * std_y)
         """
+        if not PANDAS_GT_150 and numeric_only is not no_default:
```

Comment:
This part is used in several places, might want to move it into the decorator.

Reply:
The decorator is too complicated right now imo. I’d prefer keeping this here and refactoring when I am through